Source/DafnyLanguageServer/Language/Symbols/DafnyLangSymbolResolver.cs

```diff
@@ -23,54 +22,41 @@ public class DafnyLangSymbolResolver : ISymbolResolver {
     private readonly ILogger logger;
     private readonly ILogger<CachingResolver> innerLogger;
     private readonly SemaphoreSlim resolverMutex = new(1);
-    private readonly ITelemetryPublisher telemetryPublisher;
+    private readonly TelemetryPublisherBase telemetryPublisher;
 
-    public DafnyLangSymbolResolver(ILogger<DafnyLangSymbolResolver> logger, ILogger<CachingResolver> innerLogger, ITelemetryPublisher telemetryPublisher) {
+    public DafnyLangSymbolResolver(ILogger<DafnyLangSymbolResolver> logger, ILogger<CachingResolver> innerLogger, TelemetryPublisherBase telemetryPublisher) {
       this.logger = logger;
       this.innerLogger = innerLogger;
       this.telemetryPublisher = telemetryPublisher;
     }
 
     private readonly ResolutionCache resolutionCache = new();
-    public CompilationUnit ResolveSymbols(DafnyProject project, Program program, CancellationToken cancellationToken) {
+    public void ResolveSymbols(Compilation compilation, Program program, CancellationToken cancellationToken) {
```

Comment:
It feels redundant to be passing in a program separately from the compilation here (and probably other similar places). Does it have to be the same as `compilation.programAfterParsing`?